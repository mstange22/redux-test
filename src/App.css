.app {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  flex-wrap: wrap;
  height: 100%;
}

.top,
.bottom {
  display: flex;
  height: 40%;
  align-items: center;
  width: 100%;
}

.middle {
  height: 20%;
}

.addition-container,
.subtraction-container,
.multiplication-container,
.division-container {
  width: 50%;
}

.user-input-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 16px;
}

.input-container {
  display: flex;
  flex-direction: column;
  height: 34px;
  width: 25%;
  margin-top: 5px;
}

.user-input {
  height: 100%;
  margin-left: 5px;
  margin-right: 5px;
  text-align: right;
  padding-right: 5px;
  font-size: 28px;
}