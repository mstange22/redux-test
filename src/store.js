import { createStore } from 'redux';

// initial state
const initialState = {
  a: 0,
  b: 0,
};

// constants
const SET_VALUE_A = 'SET_VALUE_A';
const SET_VALUE_B = 'SET_VALUE_B';

// actions
export const setValueA = a => ({
  type: SET_VALUE_A,
  payload: {
    a,
  },
});

export const setValueB = b => ({
  type: SET_VALUE_B,
  payload: {
    b,
  },
});

// reducer
const reducer = (state = initialState, action = {}) => {
  // console.log(action);
  switch (action.type) {
    case SET_VALUE_A:
      return { ...state, a: action.payload.a };
    case SET_VALUE_B:
      return { ...state, b: action.payload.b };
    default:
      return state;
  }
};

const store = createStore(reducer);

export default store;
