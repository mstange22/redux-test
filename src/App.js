import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './App.css';
import UserInput from './UserInput';
import Addition from './Addition';
import Subtraction from './Subtraction';
import Multiplication from './Multiplication';
import Division from './Division';

const PopUp = ({ title, message }) => {
  console.log(title);
  return (
    <div
      style={{
        textAlign: 'left',
        height: 100,
        width: 400,
        position:
        'fixed',
        top: 200,
        left: '50%',
        marginLeft: -220,
        backgroundColor: '#f4f442',
        border: 'solid 1px black',
        borderRadius: 10,
        display: 'flex',
        justifyContent: 'flex-start',
        // justifyContent: 'center',
        // alignItems: 'center',
        padding: 20,
        flexDirection: 'column',
      }}
    >
      <div className="popup-title">
        {`Pop-Up Says "${title}"`}
      </div>
      <div className="popup-message">
        {message}
      </div>
    </div>
  );
};

PopUp.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showPopUp: false,
    };
  }

  renderPopUp() {
    const { showPopUp } = this.state;
    if (!showPopUp) return null;
    return <PopUp title="Hello" message="This is a test" />;
  }

  render() {
    // const { showPopUp } = this.state;
    return (
      <div className="app">
        { this.renderPopUp() }
        <div className="top">
          <Addition />
          <Subtraction />
        </div>
        <div className="middle">
          {/* <button type="submit" onClick={() => this.setState({ showPopUp: !showPopUp })}>
            Pop-Up
          </button> */}
          <UserInput />
        </div>
        <div className="bottom">
          <Multiplication />
          <Division />
        </div>
      </div>
    );
  }
}

export default App;
