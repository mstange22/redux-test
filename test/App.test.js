import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import store from '../src/redux/store';
import App from '../src/App';

Enzyme.configure({ adapter: new Adapter() });
const shallowComponent = shallow(<App />);

describe('<App />', () => {
  test('has button', () => {
    expect(shallowComponent.exists('button')).toBe(true);
  });
  test('snapshot test', () => {
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
