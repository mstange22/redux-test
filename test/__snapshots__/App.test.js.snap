// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot test 1`] = `
<div
  className="app"
>
  <div
    className="top"
  >
    <div
      className="addition-container"
    >
      <h2>
        Addition
      </h2>
      <h2>
        0
      </h2>
      <h2>
        +
      </h2>
      <h2>
        0
      </h2>
      <h2>
        =
      </h2>
      <h2
        className="result"
      >
        0
      </h2>
    </div>
    <div
      className="subtraction-container"
    >
      <h2>
        Subtraction
      </h2>
      <h2>
        0
      </h2>
      <h2>
        -
      </h2>
      <h2>
        0
      </h2>
      <h2>
        =
      </h2>
      <h2>
        0
      </h2>
    </div>
  </div>
  <div
    className="middle"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      Show Pop-Up
    </button>
    <br />
    <br />
    <div
      className="user-input-container"
    >
      <b>
        Redux Calculator
      </b>
      <br />
      Enter two values
      <div
        className="input-container"
      >
        <input
          autoComplete="off"
          autoFocus={true}
          className="user-input"
          name="a"
          onChange={[Function]}
          value={0}
        />
        <br />
        <input
          autoComplete="off"
          className="user-input"
          name="b"
          onChange={[Function]}
          value={0}
        />
      </div>
    </div>
  </div>
  <div
    className="bottom"
  >
    <div
      className="multiplication-container"
    >
      <h2>
        Multiplication
      </h2>
      <h2>
        0
      </h2>
      <h2>
        *
      </h2>
      <h2>
        0
      </h2>
      <h2>
        =
      </h2>
      <h2>
        0
      </h2>
    </div>
    <div
      className="division-container"
    >
      <h2>
        Division
      </h2>
      <h2>
        0
      </h2>
      <h2>
        /
      </h2>
      <h2>
        0
      </h2>
      <h2>
        =
      </h2>
      <h2>
        Infinity
      </h2>
    </div>
  </div>
</div>
`;
